### ISSUE TITLE
MacOS remote desktop problem

### ISSUE THREAD

**Issue by GRIFFCOMMca** [Ylianst/MeshCentral#576]
Server = v0.4.1.p
Client = not sure (fairly late)
Client OS = Catalina i believe (cant check at the moment as its transfering data).

There are a number of issues with the MAC client, ive listed them all here as i believe they are all connected (mostly).

- Connect, most times (25% working, 75% not) only get half a screen (see image), however the mouse DOES still move on the MAC  when moving via the remote desktop view (however the remote desktop view never updates when items are clicked so cant be used )
- The DOCK when magnify is enabled, this doesn't show up on the remote desktop view as the mouse is over it, however when icons are clicked they do work
- Applications have mouse integraiton issues (specifically is KODI is on the MAC, when moving the mouse around on the remote desktop it doesnt move in KODI, however the input boxes work (so click something in KODI in on the MAC, get a text box up, that text can be entered from the remote desktop (and the remote desktop show this).
- Wake PC doesnt fully work on the MAC, that being said it does appear to wake the MAC< but only get a black screen, this maybe connected to the above screen update issue !

The above seems to indicate there is a screen capture issue on the MAC and getting the data over to the Webserver.

Many Thanks
![2019-10-17 Mac screen](https://user-images.githubusercontent.com/54428902/66978477-19235b00-f078-11e9-8f45-90a6fc4893cd.JPG)

**Comment by krayon007** [comment]
It will take some time to sort this out, as when I tried to update my Macbook Pro to Catalina, it bricked itself and refused to boot, so I have to fix my MacOS dev environment before I can get to these issues, so please bear with me..

**Comment by GRIFFCOMMca** [comment]
Hi Bryan

Thanks i have 2 MACs here (Mac Mini (mid 2012) and MacBook Pro, i will be updating the Mac Mini to 16Gb RAM (from 4Gb) and a 1Tb SSD later this week or week after (hoping that fixes the below Microsoft issue)

Mine upgraded ok (well the Mini did, not done the MacBook yet), the data transferred in ok, however Microsoft Office appears bricked (stuck on updates), kinda ironic, that the only software that didnt work on the Mac was Microsoft and THE UDPATES of all things !!!

The screen issue appears random, if its disconnected then reconnected again over and over, it will eventually  work.  Its not a LAN Issue as all the Windows machines work fine (it has a wired connection, not wifi)

**Comment by hunter-nl** [comment]
This issue exist already for some time: #377

**Comment by macdanger** [comment]
I also got the problem with 3 different Macs, all on High Sierra. For a split second, you see up to the half of the sceen when usind remote desktop but then it just disconnects without any error on the agent nor the server. I posted this in the agent issues (https://github.com/Ylianst/MeshAgent/issues/30). The rest of functions seem to work without problems.
I would really like to help analysing this but I don't know how to switch on more debug e.g. with parameters in the .msh file. Is there a documentation of the parameters in this file?
